scanner/src/funTest/kotlin/storages/AbstractStorageFunTest.kt

```diff
@@ -145,12 +147,18 @@ abstract class AbstractStorageFunTest(vararg listeners: TestListener) : WordSpec
                 val addResult = storage.add(id1, scanResult)
                 val readResult = storage.read(id1)
 
-                addResult.shouldBeFailure()
-                addResult.error shouldBe "Not storing scan result for '${id1.toCoordinates()}' because no provenance " +
-                        "information is available."
+                addResult.shouldBeFailure {
+                    it.shouldNotBeNull {
+                        message shouldBe "Not storing scan result for '${id1.toCoordinates()}' because no provenance " +
+                            "information is available."
+                    }
+                }
 
-                readResult.shouldBeSuccess()
-                readResult.result should beEmpty()
+                readResult.shouldBeSuccess {
+                    it.shouldNotBeNull {
```

Comment:
Why is this sometimes required and sometimes not?

Reply:
It depends on whether a member of `it` or only `it` itself is accessed. I've fixed this in Kotest, see https://github.com/kotest/kotest/pull/2762.